### PR TITLE
fix(argo-rollouts): controller replicas count comparison

### PR DESCRIPTION
**Pull request by jmeridth**
Fixes: #2308

Cast to int since default from values.yaml is int64 not int

<details><summary>Before</summary>
<p>

```
➜  helm install argo-rollouts . -n argo-rollouts --set controller.replicas=2 --dry-run                                                                                     

Error: INSTALLATION FAILED: template: argo-rollouts/templates/controller/deployment.yaml:51:15: executing "argo-rollouts/templates/controller/deployment.yaml" at <gt .Values.controller.replicas 1.0>: error calling gt: incompatible types for comparison
```
</p>
</details> 

<details><summary>After</summary>
<p>

```
# Source: argo-rollouts/templates/controller/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: argo-rollouts
  namespace: "argo-rollouts"
  labels:
    app.kubernetes.io/component: rollouts-controller
    helm.sh/chart: argo-rollouts-2.32.1
    app.kubernetes.io/name: argo-rollouts
    app.kubernetes.io/instance: argo-rollouts
    app.kubernetes.io/version: "v1.6.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: argo-rollouts
spec:
  selector:
    matchLabels:
      app.kubernetes.io/component: rollouts-controller
      app.kubernetes.io/name: argo-rollouts
      app.kubernetes.io/instance: argo-rollouts
  strategy:
    type: Recreate
  replicas: 2
  template:
    metadata:
      labels:
        app.kubernetes.io/name: argo-rollouts
        app.kubernetes.io/instance: argo-rollouts
        app.kubernetes.io/component: rollouts-controller
    spec:
      serviceAccountName: argo-rollouts
      containers:
      - image: "quay.io/argoproj/argo-rollouts:v1.6.0"
        args:
        - --healthzPort=8080
        - --metricsport=8090
        - --leader-elect   <---- THIS IS THE IMPORTANT BIT
        imagePullPolicy: IfNotPresent
```

</p>
</details> 

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-rollouts/templates/controller/deployment.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ spec:
         {{- if not .Values.clusterInstall }}
         - --namespaced
         {{- end }}
-        {{- if gt .Values.controller.replicas 1.0 }}
+        {{- if gt .Values.controller.replicas 1 }}
```
Comment (yu-croco) [inline charts/argo-rollouts/templates/controller/deployment.yaml:51]: On my local, I got error. 😢 

```
$ helm template . > foo.yaml

Error: template: argo-rollouts/templates/controller/deployment.yaml:51:15: executing "argo-rollouts/templates/controller/deployment.yaml" at <gt .Values.controller.replicas 1>: error calling gt: incompatible types for comparison
```

Comment (jmeridth) [reply charts/argo-rollouts/templates/controller/deployment.yaml]: Fixed

**Review by yu-croco — APPROVED** [review]
Thank you!🚀

File: charts/argo-rollouts/templates/controller/deployment.yaml (hunk 1 above)
Comment (jmeridth) [reply charts/argo-rollouts/templates/controller/deployment.yaml]: What's nice is our [CI caught it[(https://github.com/argoproj/argo-helm/actions/runs/6625851696/job/17997619786#step:7:46). 😄 

Ran that locally and was able to reproduce. CI ftw.

**Review by tico24 — APPROVED** [review]

**Pull request merged**